packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -37,6 +37,8 @@ const EMPTY_DATA = {};
  * @return {Object} Block Editor Settings.
  */
 function useBlockEditorSettings( settings, hasTemplate ) {
+	const richDataLocalCache = useRef( {} );
```

Comment:
Debugging this will be weird as the place of this cache object is unexpected. This is also because:

```
// Temporary home - should this live in `core-data`?
```

I think so yes. As `fetchRemoteUrlData` this looks like a special case of this one. Then caching could be added at the origin of the data in `fetchRemoteUrlData`.

Reply:
I agree that maybe the caching could be added in `fetchRemoteUrlData`. Also I would probably choose to use a `Map` than an object, for readability.

Another option would be to add HTTP caching on backend, it feels more _natural_ to do that too. One possible downside would be that it'll still fire a request, so the UI will probably see a small flickering when changing `url`, not sure if we need to optimize that though.

Reply:
We don't have out of the box ways to cache serverside across requests.

Reply:
Http caching is already in place server side. This is purely to avoid another http request on the client side.

Reply:
> Http caching is already in place server side.

Is it? When testing in gutenberg.run, it seems like the response header is returning `Cache-Control: no-cache, must-revalidate, max-age=0`. Might just be a gutenberg.run issue though.

Do you know what the cache setting is for this api? I'm thinking maybe the `max-age` could be larger than normal requests, like maybe for 6 hours or 1 day?

Reply:
Oh sorry I was confused. I meant we are caching the response in a transient at a WP level and then serving that cached data for requests for the same remote URL.

Not sure if we're implementing HTTP caching. How do we even control that at a WP level? Isn't it generally a sever config thing? Maybe @hellofromtonya would know?

Reply:
I think it's still possible since that PHP is a server side language? I know very little about PHP though 😅.